.github/workflows/manual-sync.yml

```diff
@@ -0,0 +1,95 @@
+name: Sync from osu! upstream
+
+# Workflow dispatch can take inputs like any other github.event
+# As part of #5090 - we should evaluate adding inputs and outputs to refine this.
+on:
+  workflow_dispatch:
+    inputs:
+      backup:
+        description: "Create a backup of your target branch? (true/false)"
+        required: true
+        default: "false"
+      force:
+        description: "Overwrite any changes in the target repository? (true/false)"
+        required: true
+        default: "false"
+
+jobs:
+  sync:
+    name: Execute Sync Task
+    runs-on: ubuntu-latest
+    steps:
+     # WARNING: DO NOT USE SHALLOW CLONES FOR THIS STEP! Doing so will result on git merge thinking we diverged from upstream!
+      - name: Checkout
+        shell: bash
+        run: |
+          REPOSITORY="https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git"
+          # merge commit ref name (with refs/heads/ stripped out)
+          BRANCH="${GITHUB_REF/#refs\/heads\//}"
+
+          git version
+          # clone without blobs; don't checkout to avoid fetching them anyway
+          git clone --filter=blob:none --no-checkout ${REPOSITORY} .
+          git config --local gc.auto 0
+
+          # set up sparse checkout
+          git sparse-checkout init
+          git sparse-checkout set '**/*.md' '**/.remark*' '**/*.json' '**/*.yaml'
+
+          git checkout --progress --force -B $BRANCH refs/remotes/origin/$BRANCH
```

Comment:
What's with the unexplained sparse checkout that wasn't here before? I don't think it needs to be used here. It's only used in wiki CI to speed it up by avoiding fetching blobs that don't need to be fetched (because they're not inspected anyway).

This case is different, as pretty much any blob changed on a topic branch will conflict. What will happen in that case if this checkout is sparse?

No reason to not go with a default checkout, I'd say.

Reply:
Right, this was a haphazard copy after 3 hours of bashing my head because `actions/checkout` was breaking the pipeline continuously. Good catch.